lib/generators/json_serializable_generator.dart

```diff
@@ -223,14 +229,58 @@ class JsonSerializableGenerator
         // ...resetting `isList` to `false`.
         isList = false;
       }
-    }
 
-    if (!isList && _coreIterableChecker.isAssignableFromType(fieldType)) {
-      // Then we need to add `?.toList()
-      expression += "?.toList()";
+      if (!isList) {
+        // Then we need to add `?.toList()
+        expression += "?.toList()";
+      }
+
+      return expression;
+    } else if (_coreMapChecker.isAssignableFromType(fieldType)) {
+      var args = _getTypeArguments(fieldType, _coreMapChecker);
```

Comment:
nit: this method is getting pretty long so it might be worth breaking out each of these sections into their own methods.